README.md

```diff
@@ -1521,6 +1521,35 @@ To run the tests you will need:
 $ mvn test
 ```
 
+## Versioning
+
+This library follows [Semantic Versioning](http://semver.org/), but with some
+additional qualifications:
+
+1. Components marked with `@BetaApi` are considered to be "0.x" features inside
+   a "1.x" library. This means they can change between minor and patch releases
+   in incompatible ways. These features should not be used by any library "B"
+   that itself has consumers, unless the components of library B that use
+   `@BetaApi` features are also marked with `@BetaApi`. Features marked as
+   `@BetaApi` are on a path to eventually become "1.x" features with the marker
+   removed.
+
+1. Components marked with `@InternalApi` are technically public, but are only
+   public for technical reasons, because of the limitations of Java's access
+   modifiers. For the purposes of semver, they should be considered private.
+
+1. Components marked with `@InternalExtensionOnly` are stable for usage, but
+   not for extension. Thus, methods will not be removed from interfaces marked
+   with this annotation, but methods can be added, thus breaking any
+   code implementing the interface. See the javadocs for more details on other
+   consequences of this annotation.
+   
+1. Components marked with `@ObsoleteApi` are stable for usage in the current major version,
+   but will be marked with `@Deprecated` in a future major version.
+   **NOTE**: We reserve the right to mark anything as `@Deprecated` and introduce breaking
+   changes in a minor version for anythign that we deem to be ***critical bugs and
```

Comment:
```suggestion
   changes in a minor version to fix any ***critical bugs and
```

Reply:
Can you also move the section for InternalApi and ObsoleteApi/ Deprecated to the top of the section? I think those two are a far more important for readers